language/functional_tests/tests/testsuite/validator_set/add_validator.mvir

```diff
@@ -0,0 +1,79 @@
+//! account: alice
+//! account: vivian, 1000000, 0, validator
+
+// add_validator cannot be called on an account without a ValidatorConfig resource
+//! sender: association
+import 0x0.LibraSystem;
+import 0x0.ValidatorConfig;
+main() {
+    assert(!ValidatorConfig.has({{alice}}), 70);
+    LibraSystem.add_validator({{alice}});
+
+    return;
+}
+
+// check: ABORTED
+// check: 17
+
+//! new-transaction
+//! sender: association
+// add_validator cannot be called on an existing validator
+import 0x0.LibraSystem;
+main() {
+    LibraSystem.add_validator({{vivian}});
+
+    return;
+}
+
+// check: ABORTED
+// check: 18
+
+//! new-transaction
+//! sender: alice
+import 0x0.ValidatorConfig;
+main() {
+    ValidatorConfig.register_candidate_validator(h"10", h"20", h"30");
```

Comment:
anyone can register as candidate?

Reply:
Yes. Registering as a candidate is necessary, but not sufficient to become a validator. The Association account needs to call `add_validator` on an address that's already registered.